### PR TITLE
Make CResourceMonitor report peak model memory usage

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR is a complement of the Java PR (https://github.com/elastic/elasticsearch/pull/59017).
It makes `CResourceMonitor` class report peak model memory usage.

Closes https://github.com/elastic/elasticsearch/issues/59130

### CODE REVIEW

File: lib/model/CResourceMonitor.cc [hunk 1]
```diff
@@ -58,7 +58,13 @@ void CResourceMonitor::unRegisterComponent(CMonitoredResource& resource) {
 
     m_MonitoredResourceCurrentMemory -= itr->second;
     m_Resources.erase(itr);
-    core::CProgramCounters::counter(counter_t::E_TSADMemoryUsage) = this->totalMemory();
+    std::size_t total{this->totalMemory()};
+    core::CProgramCounters::counter(counter_t::E_TSADMemoryUsage) = total;
+    core::CProgramCounters::counter(counter_t::E_TSADPeakMemoryUsage) = total;
```
Comment (droberts195) [inline lib/model/CResourceMonitor.cc:63]: It looks like this line is a mistake.  It will cause the peak to decrease.

**Comment by przemekwitek** [comment]
> It would be nice to have a test that proves that peak memory usage is remembered between persisting and restoring.

Done in `CMetricAnomalyDetectorTest`

File: docs/CHANGELOG.asciidoc [hunk 2]
```diff
@@ -67,6 +67,7 @@
 * Improve runtime and memory usage training deep trees for classification and
   regression. (See {ml-pull}1340[#1340].)
 * Improvement in handling large inference model definitions. (See {ml-pull}1349[#1349].)
+* Make CResourceMonitor class report peak_model_bytes field. (See {ml-pull}1389[#1389].)
```
Comment (droberts195) [inline docs/CHANGELOG.asciidoc:70]:
```suggestion
* Add a peak_model_bytes field to model_size_stats. (See {ml-pull}1389[#1389].)
```

Comment (przemekwitek) [reply docs/CHANGELOG.asciidoc]: Done.

**Pull request merged**